readinglist/__init__.py

```diff
@@ -78,7 +84,12 @@ def on_new_request(event):
     config.add_subscriber(on_new_request, NewRequest)
 
     def on_new_response(event):
-        # Display the status of the request as well as the time spend
+        if hasattr(event.request, '_received_at'):
+            duration = (msec_time() - event.request._received_at)
+        else:
+            duration = "unknown"
```

Comment:
Well, it does! When you create a response directly then the event is triggered and the "event.request" associated hasn't passed the on_new_request (because, it's not a new request).